scanner/Makefile

```diff
@@ -113,60 +122,221 @@ ifdef CI
 	}
 endif
 	$(SILENT)go mod verify
-	$(SILENT)touch deps
+	$(SILENT)touch $@
 
 .PHONY: ossls-notice-no-download
 ossls-notice-no-download: deps
-	$(SILENT)echo "+ $@"
+	@echo "+ $@"
 	$(SILENT)ossls version
 	$(SILENT)ossls audit --export image/scanner/THIRD_PARTY_NOTICES
 
 .PHONY: ossls-notice
 ossls-notice: deps $(OSSLS_BIN)
-	$(SILENT)echo "+ $@"
+	@echo "+ $@"
 	$(SILENT)$(OSSLS_BIN) version
 	$(SILENT)$(OSSLS_BIN) audit --export image/scanner/THIRD_PARTY_NOTICES
 
 SCRIPTS := restore-all-dir-contents import-additional-cas save-dir-contents
 .PHONY: copy-scripts
 copy-scripts: $(addprefix ../image/rhel/static-bin/,$(SCRIPTS))
-	$(SILENT)echo "+ $@"
+	@echo "+ $@"
 	$(SILENT)cp $^ image/scanner/scripts
 
 .PHONY: copy-db-scripts
 copy-db-scripts: ../image/postgres/download.sh
-	$(SILENT)echo "+ $@"
+	@echo "+ $@"
 	$(SILENT)cp $^ image/db
 
 .PHONY: copy-db-image-scripts
 copy-db-image-scripts: ../image/postgres/scripts/init-entrypoint.sh
-	$(SILENT)echo "+ $@"
+	@echo "+ $@"
 	$(SILENT)cp $^ image/db/scripts
 
 .PHONY: copy-db-sigs
 copy-db-sigs: ../image/postgres/signatures
-	$(SILENT)echo "+ $@"
+	@echo "+ $@"
 	$(SILENT)cp -R $^ image/db/signatures
 
+##################
+## Certificates ##
+##################
+
+# Certificates vars
+# =================
+
+SCANNER_CERTS_NAMESPACE ?= $(SCANNER_NAMESPACE)
+
+certs-d         := certs
+certs-names     := stackrox-v4 stackrox-v4-db scannerctl
+certs-namespace := $(SCANNER_CERTS_NAMESPACE)
+
+# Targets.
+certs-t := $(foreach name,$(certs-names),$(certs-d)/$(name)/.verified)
+
+# Subject prefix.
+certs-subj := /C=US/ST=North Carolina/L=Raleigh/O=Red Hat, Inc./OU=Stackrox Scanner
+
+# DNS name or CN of the current cert target.
+cert-dns = $(notdir $(@D)).$(certs-namespace)
+
+# Certificates rules
+# ==================
+
+.PHONY: certs clean-certs
+
+certs: $(certs-t)
+
+clean-certs:
+	@echo "+ $@"
+	$(SILENT)rm -rf certs/
+
+# Generate keys.
+#
+.PRECIOUS: $(certs-d)/%/key.pem $(certs-d)/%/cert.pem
+
+$(certs-d)/%/key.pem:
+	@echo "+ $@"
+	$(SILENT)mkdir -p $(@D)
+	$(SILENT) openssl genpkey  \
+	    -algorithm RSA \
+	    -out $@
+
+# Generate root certificate authority.
+#
+$(certs-d)/ca/root.pem: $(certs-d)/ca/key.pem
+	@echo "+ $@"
+	$(SILENT) openssl req \
+	    -new \
+	    -x509 \
+	    -nodes \
+	    -days 398 \
+	    -subj "$(certs-subj)/CN=RootCA" \
+	    -key $< \
+	    -out $@
+
+# Generate a certificate signing request.
+#
+$(certs-d)/%/cert.csr: $(certs-d)/%/key.pem
+	$(SILENT) openssl req \
+	    -new \
+	    -subj "$(certs-subj)/CN=$(cert-dns)" \
+	    -key $< \
+	    -out $@
+
+# Generate a certificate.
+#
+.PRECIOUS: $(certs-d)/%/ca.pem
+$(certs-d)/%/ca.pem: $(certs-d)/ca/root.pem
+	@echo "+ $@"
+	$(SILENT)cp $< $@
+
+$(certs-d)/%/cert.pem: $(certs-d)/%/cert.csr $(certs-d)/%/ca.pem $(certs-d)/ca/key.pem
```

Comment:
There may still be an issue here:

```
$ make -C scanner e2e-deploy
mkdir -p e2etests/helmchart/files
cp image/db/postgresql.conf e2etests/helmchart/files/db-postgresql.conf
mkdir -p e2etests/helmchart/files
cp image/db/pg_hba.conf e2etests/helmchart/files/db-pg_hba.conf
+ certs/ca/key.pem
mkdir -p certs/ca
2>/dev/null openssl genpkey  \
	    -algorithm RSA \
	    -out certs/ca/key.pem
+ certs/ca/root.pem
openssl req \
	    -new \
	    -x509 \
	    -nodes \
	    -days 398 \
	    -subj "/C=US/ST=North Carolina/L=Raleigh/O=Red Hat, Inc./OU=Stackrox Scanner/CN=RootCA" \
	    -key certs/ca/key.pem \
	    -out certs/ca/root.pem
cp certs/ca/root.pem e2etests/helmchart/files/ca.pem
+ certs/scanner-v4/key.pem
mkdir -p certs/scanner-v4
2>/dev/null openssl genpkey  \
	    -algorithm RSA \
	    -out certs/scanner-v4/key.pem
openssl req \
	    -new \
	    -subj "/C=US/ST=North Carolina/L=Raleigh/O=Red Hat, Inc./OU=Stackrox Scanner/CN=scanner-v4.stackrox" \
	    -key certs/scanner-v4/key.pem \
	    -out certs/scanner-v4/cert.csr
+ certs/scanner-v4/ca.pem
cp certs/ca/root.pem certs/scanner-v4/ca.pem
+ certs/scanner-v4/cert.pem
2>/dev/null openssl x509 \
	    -req \
	    -days 365 \
	    -in certs/scanner-v4/cert.csr \
	    -CA certs/scanner-v4ca.pem \
	    -CAkey certs/ca/key.pem \
	    -CAcreateserial \
	    -sha256 \
	    -extfile <(echo "subjectAltName = DNS:scanner-v4.stackrox DNS:localhost DNS:127.0.0.1") \
	    -out certs/scanner-v4/cert.pem
make: *** [certs/scanner-v4/cert.pem] Error 1
rm certs/scanner-v4/ca.pem certs/scanner-v4/cert.csr
```

Reply:
To add another data point to this: if the cwd is `scanner`, `make e2e-deploy` seems to work

Reply:
I can't reproduce it, are you guys working on top of 082af453bf?

```
⮩  make -C scanner e2e-deploy
make: Entering directory '/home/jvdm/wt/github.com/stackrox/stackrox/scannerv4-e2e/scanner'
+ certs/ca/key.pem
..+........+....+...+..+......+...+.........+............+....+..........................+...+.+.....+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.+............+.............+......+.........+.....+......+................+......+..+............+....+..+...+.+.....+......+...+....+...+........+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*............+...+.......+.........+...+..+....+..+..................+.+...+...................................+...+...+.+.....+....+.........+...+.........+.....+...+.+.....+.+......................................+.+..+.........+.+.....+....+.........+...........+....+.........+......+......+...+...+........+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
.+...+..+.........+.+...+...............+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*..+.+..+...+....+...+..+...+.+..+..........+...+..+...+.+...+...........+.........+...+...+......+.+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*..........+........+.+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+ certs/ca/root.pem
+ certs/scanner-v4/key.pem
.......+.......+..+.............+..+.+.....+.......+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.......+...+..................+......+.+..+......+....+.....+...+....+.....+.+.....+.......+...+...+..+....+.....+....+..+...+................+...+...+..+.......+.....+.+...........................+..+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.+...+............+.....+.............+......+...+...+..+.......+...+.....+.......+......+...+.....+....+............+...+..+......+...+......+....+..+............+.+.....+.+...+............+........+.............+...+.....+....+........+.........+.+........+................+...+.....+.......+.........+..+.........+....+.....+.........+.+..+.........+.+..+......+.......+........+......+.............+...............+........+...+......+.+...+.....+.+........+.....................+......+....+..+..........+.....+....+.........+..+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
..+.+.........+.....+......+.+.....+....+...+.....+...............+.+.........+..+.+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*...+.....+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.......+....+...+.....+.........+......+.........+.............+..+...+.......+...+...+..+.+..+....+...+...+..+.......+........+.+..+....+........+.....................+....+...+.....+.......+..+.........+.+...+.....+......+.........+.+............+..+..........+.....+...+...+.+.....+....+...........+......+....+...+............+......+..+.....................+.+......+.....+.........+.........+.+.........+.....+......+.+............+......+..+....+...+.....+...+.........+.+..+......+.......+........+.+......+........+.+...+..+.........+............+.......+...........+.......+..+.+......+.........+......+.....+..........+...+...+.....+....+....................+...+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+ certs/scanner-v4/ca.pem
+ certs/scanner-v4/cert.pem
Certificate request self-signature ok
subject=C = US, ST = North Carolina, L = Raleigh, O = "Red Hat, Inc.", OU = Stackrox Scanner, CN = scanner-v4.stackrox
certs/scanner-v4/cert.pem: OK
+ certs/scanner-v4-db/key.pem
..+....+.....+.........+.+.........+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*..+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.+.........+..........+........+.+......+...........+....+...+......+...........+..........+........+......+.........+...+...+....+...+...+............+..+.+.....+.........+............+.+...+..+...................+..................+...........+...............+.+......+........+....+...+..+......+......+.+..............+..........+.....+......+...+................+..............+......+......+....+...+..+...+............................+..+...+....+.....+.........+.+........+.+......+......+..+.+.....+..........+..+.......+..+...+...+...+....+...+...+.....+....+...+.........+..................+.....+...+............+.+.................+.......+.....+...+...............+.............+..+....+.....+....+.....+.+...+...............+..+...+.+...........+.+.....+...+.+.....+..........+...+...+............+...............+........+..........+............+.....+.+......+......+.....+.......+..+...................+..+.........+..........+...+...........+............+.........+.+..+......+......+...+.......+......+..+.......+.....+.........+.........+.+..+....+........+...+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
..+.+........+....+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*...+....+........+....+.........+..+.......+...+..+.+......+.....+...+.+............+..+...+.......+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.+...............+..+.+.....+.+...+................................+....+..+....+...........+....+...........+...+.......+.........+......+.....+....+.....+...+......+.+.....+......+............+...............+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+ certs/scanner-v4-db/ca.pem
+ certs/scanner-v4-db/cert.pem
Certificate request self-signature ok
subject=C = US, ST = North Carolina, L = Raleigh, O = "Red Hat, Inc.", OU = Stackrox Scanner, CN = scanner-v4-db.stackrox
certs/scanner-v4-db/cert.pem: OK
+ e2e-deploy 4.2.x-277-g082af453bf
Error from server (AlreadyExists): namespaces "stackrox" already exists
make: [Makefile:294: e2e-deploy] Error 1 (ignored)
Release "scanner-v4-e2e" does not exist. Installing it now.
NAME: scanner-v4-e2e
LAST DEPLOYED: Thu Oct  5 13:06:01 2023
NAMESPACE: stackrox
STATUS: deployed
REVISION: 1
TEST SUITE: None
rm certs/scanner-v4/cert.csr certs/scanner-v4-db/cert.csr
make: Leaving directory '/home/jvdm/wt/github.com/stackrox/stackrox/scannerv4-e2e/scanner'
[infra/jvdm-sockeye] ⮩  kubectl -n stackrox get pod all                                                 [~/wt/github.com/stackrox/stackrox/scannerv4-e2e]
Error from server (NotFound): pods "all" not found
⮩  kubectl -n stackrox get pod
NAME                             READY   STATUS              RESTARTS   AGE
scanner-v4-b844548b7-l422v       0/1     ContainerCreating   0          8s
scanner-v4-db-5bf95459bc-wssj6   0/1     Init:0/1            0          8s
⮩  
```